core/src/main/scala/org/bitcoins/core/gcs/BlockFilterMatcher.scala

```diff
@@ -0,0 +1,128 @@
+package org.bitcoins.core.gcs
+
+import org.bitcoins.core.number.UInt64
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+sealed trait BlockFilterMatcher {
+
+  /**
+    * Checks if the underlying filter matches the given data
+    */
+  def matches(data: ByteVector): Boolean
+
+  /**
+    * Checks if the underlying filter matches any item from the given collection
+    */
+  def matchesAny(data: Vector[ByteVector]): Boolean
```

Comment:
Couldn't we implement this here? I.e. 

```scala
def matchesAny(data: Vector[ByteVector]): Boolean = {
  data.exists(d => matches(d))
}
```

Reply:
What would be a benefit? The current recommended implementation doesn't work this way. And we don't know how other filter types will be implemented.

Reply:
Normal "good" software engineering practices. Also, with `.exists()` i believe we exit early if we match something, so we don't have to evaluate the entire sequence of elements

Reply:
I'd rather leave `BlockFilterMatcher` trait fully abstract, so that we don't impose the algorithm on the future implementations.